orm/src/test/java/org/hibernate/search/test/jpa/SpatialQueryingJPATest.java

```diff
@@ -226,6 +227,63 @@ public void testDistanceSort2() throws Exception {
 	}
 
 	@Test
+	@TestForIssue(jiraKey = "HSEARCH-2322")
+	public void testDistanceSortMissingCoordinatesWholeSegment() throws Exception {
+		POI poi = new POI( 0, "Distance to 24,32 : 10.16km", 24.0d, 31.9d, "" );
+		POI poi2 = new POI( 1, "Distance to 24,32 : unknown, 4361.00km if interpreted as 0,0", null, null, "" );
+
+		FullTextEntityManager em = Search.getFullTextEntityManager( factory.createEntityManager() );
+
+		em.getTransaction().begin();
+		em.persist( poi );
+		em.getTransaction().commit();
+
+		/*
+		 * Create the POI with a missing value in a separate transaction, so that
+		 * the document will be alone in one segment (or so it seems...?)
+		 */
+		em.getTransaction().begin();
+		em.persist( poi2 );
+		em.getTransaction().commit();
+
+		em.getTransaction().begin();
+		double centerLatitude = 24.0d;
+		double centerLongitude = 32.0d;
+
+		final QueryBuilder builder = em.getSearchFactory().buildQueryBuilder().forEntity( POI.class ).get();
+
+		org.apache.lucene.search.Query luceneQuery = builder.all().createQuery();
+
+		FullTextQuery hibQuery = em.createFullTextQuery( luceneQuery, POI.class );
+		Sort distanceSort = new Sort( new DistanceSortField( centerLatitude, centerLongitude, "location" ) );
+		hibQuery.setSort( distanceSort );
+		hibQuery.setProjection( FullTextQuery.THIS, FullTextQuery.SPATIAL_DISTANCE );
+		hibQuery.setSpatialParameters( centerLatitude, centerLongitude, "location" );
+		List results = hibQuery.getResultList();
+		Object[] firstResult = (Object[]) results.get( 0 );
+		Object[] secondResult = (Object[]) results.get( 1 );
+		Assert.assertEquals( 10.1582, (Double) firstResult[1], 0.01 );
+		// ElasticSearch returns a distance of Double.MAX_VALUE while Lucene just returns the distance to (0,0)
```

Comment:
Could you add a reference to HSEARCH-2324 in both comments so that we can keep track of it.

Other than that it looks good to me.

Thanks!